src/common/xio_connection.c

```diff
@@ -1751,6 +1751,9 @@ static void xio_fin_req_timeout(void *data)
 	/* flush all messages back to user */
 	xio_connection_notify_msgs_flush(connection);
 
+	if (connection->nexus)
+		xio_nexus_close(connection->nexus, &connection->session->observer);
```

Comment:
Is it not a problem calling `xio_nexus_close()` once again in `xio_connection_post_destroy()` if the `kref_put()` at the end of the function results in its being called?